packages/fxa-graphql-api/src/gql/account.resolver.ts

```diff
@@ -430,6 +421,27 @@ export class AccountResolver {
     input: MetricsOptInput
   ): Promise<BasicPayload> {
     await Account.setMetricsOpt(uid, input.state);
+
+    if (!['in', 'out'].includes(input.state)) {
+      throw new Error(
+        `Invalid state. Expected 'in' or 'out'. But recieved: ${input.state}`
+      );
+    }
+
+    await this.notifier.send({
+      event: 'metricsChange',
+      data: {
+        uid,
+        enabled: input.state === 'in',
+
+        // TODO: Since we fan this out, I don't think the client id is needed?
+        // clientId,
```

Comment:
That's correct, this is only interpreted as the clientId of the oauth client that generated the event, the event-broker doesn't use it.